ptf/lpm.py

```diff
@@ -0,0 +1,110 @@
+import random
+
+from ipaddress import ip_address, ip_network
+from SubnetTree import SubnetTree
+
+'''
+NOTE: This module is pulled from sonic-mgmt repo
+
+LpmDict is a class used in FIB test for LPM and IP segmentation.
+
+This class contains a class variable SubnetTree() _subnet_tree to solve the
+LPM search functionality, which is achieved using Patricia tree. In order to
+have IP segmentation functionality: segment the whole IP space into different
+segments from start to end according to the prefixes (networks) it reads.
+
+Initially, the whole IP space contains only one range. After inserting
+prefixes, the IP space is segmented into multiple ranges. The ranges()
+function returns all ranges in the LpmDict with a list of IpIntervals. The
+sub-class IpInterval then could be used to get the first/last/random IP within
+this range. It could also check the length of the range and if an IP is within
+this range.
+
+To achieve the LPM functionality, use the LpmDict as a dictionary and use
+[] operator to get the corresponding value using the key (IP).
+
+Please check the test_lpm.py file to see the details of how this class works.
+'''
+class LpmDict():
```

Comment:
This module is to be used in saiswitch tests that will be upstreamed soon. But in general it is a part of the infrastructure and may be used in any other test modules.

Reply:
Code shouldn't be added with the assumption it will be used, this way we may end up with unused code lying around. Can you remove this file and add it when it's being used, i.e. along with saiswitch tests.

Reply:
Okay, as you wish